subserver_mgr.go

```diff
@@ -0,0 +1,399 @@
+package terminal
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+// subServerCfg defines the config values that must be set in order to properly
+// define a subServer.
+type subServerCfg struct {
+	// name is the string identifier of the sub-server.
+	name string
+
+	// remote must be set if the sub-server is running remotely and so must
+	// not be started in integrated mode.
+	remote bool
+
+	// startIntegrated defines how the sub-server should be started if it
+	// is to be run in integrated mode.
+	startIntegrated func(lnrpc.LightningClient, *lndclient.GrpcLndServices,
+		bool) error
+
+	// stopIntegrated defines how the sub-server should be stopped if it is
+	// running in integrated mode.
+	stopIntegrated func() error
+
+	// registerGrpcService is a function that can be called in order to
+	// register the sub-server's rpc with the given registrar.
+	registerGrpcService func(grpc.ServiceRegistrar)
+
+	// macValidator will be used to verify any macaroons for requests that
+	// are to be handled by the sub-server.
+	macValidator macaroons.MacaroonValidator
+
+	// onStartError will be called if there is a startup error (in
+	// integrated mode) or if the remote connection could not be made in
+	// remote mode.
+	onStartError func(err error)
```

Comment:
Do we plan `onStartError`, `onStartSuccess` to do more things in the future beyond setting the status for this subserver? If not we can directly set the status using `subServerCfg.name`